In [ ]:
import pandas as pd

# Load the CSV file to inspect the data
file_path = 'daily_sales.csv'
data = pd.read_csv(file_path)

# Display the first few rows of the dataset to understand its structure
data.head()


Description: this dataset shows number of WaterCure units sold per day and the corresponding daily temperature from January 2024 to August 2024  \
0                                                NaN                                                                                                
1                                               Date                                                                                                
2                                           1/1/2024                                                                                                
3                                           1/2/2024                                                                                                
4                                           1/3/2024                                                                                                

         Unnamed: 1        Unnamed: 2             Unnamed: 3  
0               NaN               NaN                    NaN  
1  Daily Units Sold  Daily Unit Price  Daily Temperature (C)  
2                91                24                     25  
3                90                24                     24  
4                70                24                     19

In [ ]:
#the data has some headers and extra rows that need to be cleaned up

# Removing the unwanted rows
data_clean = pd.read_csv(file_path, skiprows=2)

# Rename the columns to more meaningful names
data_clean.columns = ['Date', 'Daily Units Sold', 'Daily Unit Price', 'Daily Temperature (C)']

# Remove any rows that might still contain NaN values
data_clean = data_clean.dropna()

# Convert the Date column to datetime format
data_clean['Date'] = pd.to_datetime(data_clean['Date'], format='%m/%d/%Y')

# Display the first few rows of the cleaned data
data_clean.head()


Date  Daily Units Sold  Daily Unit Price  Daily Temperature (C)
0 2024-01-01                91              24.0                     25
1 2024-01-02                90              24.0                     24
2 2024-01-03                70              24.0                     19
3 2024-01-04                89              24.0                     23
4 2024-01-05               100              24.0                     36

In [ ]:
import pandas as pd

# Calculation the summary statistics for the 'Daily Units Sold' and 'Daily Temperature (C)'
summary_stats = data_clean[['Daily Units Sold', 'Daily Temperature (C)']].describe()

# Display the summary statistics
summary_stats

Daily Units Sold  Daily Temperature (C)
count        244.000000             244.000000
mean          66.479508              19.512295
std           19.003021               5.374442
min           11.000000               7.000000
25%           57.000000              16.000000
50%           63.500000              19.000000
75%           81.250000              23.000000
max          100.000000              36.000000

#Summary Interpretation
1. Count: The dataset contains 244 daily entries for both units sold and daily temperature, spanning the period from January to August 2024.

2. Mean: The average number of WaterCure units sold per day is 66.48, and the average daily temperature is 19.51°C during the same period.

3. Standard Deviation (std): The variability in the number of units sold is 19.00 units per day, indicating moderate fluctuations in daily sales. The temperature varies by about 5.37°C daily, showing some climate changes over the months.

4. Minimum and Maximum: The minimum number of WaterCure units sold in a day is 11, and the maximum is 100 units. Regarding temperature, the minimum observed daily temperature is 7°C, and the maximum is 36°C.

In [ ]:
# Calculate the Pearson correlation coefficient between Daily Units Sold and Daily Temperature
correlation = data_clean['Daily Units Sold'].corr(data_clean['Daily Temperature (C)'])

correlation


0.8044269923525676

**The Pearson correlation coefficient between daily units sold and daily temperature is approximately 0.80.**

This indicates a strong positive correlation between the two variables, meaning that as the temperature increases, the number of WaterCure units sold tends to increase as well. This suggests that the relationship between sales and temperature is likely driven by climate, supporting the initial hypothesis that higher temperatures lead to increased sales of WaterCure.


In [ ]:
# Sort the data frame by 'Daily Units Sold' in descending order
sorted_data = data_clean.sort_values(by='Daily Units Sold', ascending=False)



In [ ]:
# save the sorted data to a CSV file named "WaterCure_analysis_Holankar_Siddharth.csv"
output_file_path = 'WaterCure_analysis_Holankar_Siddharth.csv'
sorted_data.to_csv(output_file_path, index=False)

# Return the path to the generated file
output_file_path


'WaterCure_analysis_Holankar_Siddharth.csv'